runtime/src/message_processor.rs

```diff
@@ -479,6 +490,7 @@ impl MessageProcessor {
             instruction.program_id(&message.account_keys),
             rent_collector.rent,
             pre_accounts,
+            self.programs.clone(),
```

Comment:
I'm not proud of this clone, my main goal is to unblock 1.1 and 1.2 releases by landing https://github.com/solana-labs/solana/pull/10375.

This could be one of those "non-jit BPF optimizations" in the v1.3 time frame

Reply:
Yeah, we should be able to pass a reference here, I'll file an issue to track when this lands